resolwe_bio/processes/expression/salmon.py

```diff
@@ -0,0 +1,339 @@
+            )
+
+        options = GroupField(Options, label="Options", hidden="!advanced")
+
+    class Output:
+        """Output fields."""
+
+        exp = FileField(label="Normalized expression")
+        exp_json = JsonField(label="Expression (json)")
+        exp_type = StringField(label="Expression type")
+        rc = FileField(label="Read counts", required=False)
+        exp_set = FileField(label="Expressions")
+        exp_set_json = JsonField(label="Expressions (json)")
+        quant = FileField(label="Salmon quant file")
+        salmon_output = DirField(label='Salmon output')
+        txdb = FileField(label="Transcript to gene mapping")
+        strandedness = StringField(label='Strandedness code')
+        source = StringField(label="Gene ID source")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+        feature_type = StringField(label="Feature type")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        if inputs.salmon_index.species != inputs.annotation.species:
+            self.error('Salmon index file species ({}) must match GTF annotation '
+                       'file species ({})'.format(inputs.salmon_index.species, inputs.annotation.species))
+
+        if inputs.salmon_index.build != inputs.annotation.build:
+            self.error('Salmon index file build ({}) must match GTF annotation '
+                       'file build ({})'.format(inputs.salmon_index.build, inputs.annotation.build))
+
+        if inputs.salmon_index.source != inputs.annotation.source:
+            self.error('Salmon index file source ({}) must match GTF annotation '
+                       'file source ({})'.format(inputs.salmon_index.source, inputs.annotation.source))
+
+        if inputs.options.no_length_correction:
+            abundance_unit = 'CPM'
+            output_suffix = '_cpm.txt.gz'
+        else:
+            abundance_unit = 'TPM'
+            output_suffix = '_tpm.txt.gz'
+
+        args = [
+            '-i', inputs.salmon_index.index.path,
+            '-l', inputs.options.stranded,
+            '--incompatPrior', inputs.options.incompat_prior,
+            '--minAssignedFrags', inputs.options.min_assigned_frag,
+            '--rangeFactorizationBins', inputs.options.range_factorization_bins,
+            '-p', self.requirements.resources.cores,
+            '-o', 'salmon_output',
+        ]
+
+        # Prepare .FASTQ file inputs based on the reads input type
+        if inputs.reads.type.startswith('data:reads:fastq:single:'):
+            args.extend(['-r'] + [lane.path for lane in inputs.reads.fastq])
+        else:
+            args.extend(['-1'] + [lane.path for lane in inputs.reads.fastq])
+            args.extend(['-2'] + [lane.path for lane in inputs.reads.fastq2])
+
+        # Prepare optional inputs
+        if inputs.options.seq_bias and not inputs.options.no_length_correction:
+            args.append('--seqBias')
+        else:
```

Comment:
I'm not sure that the condition to trigger this warning is correct.

See the following truth table. Discrepancy occurs in the last two rows.

|inputs.options.seq_bias|inputs.options.no_length_correction|display warning (current impl.)|display warning (desired impl.)|
|:-:|:-:|:-:|:-:|
|True|True|True|True|
|True|False|False|False|
|False|True|True|False|
|False|False|True|False|

The same applies to the next warning as well.

Reply:
Fixed in https://github.com/genialis/resolwe-bio/pull/794